Empty Function causes SegFault

**What happened?** 

```
> devspace print
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x105d20938]

goroutine 1 [running]:
github.com/loft-sh/devspace/pkg/devspace/config/versions.adjustConfig(0x140007f8690)
        /Users/runner/work/devspace/devspace/pkg/devspace/config/versions/adjust.go:26 +0x2c8
github.com/loft-sh/devspace/pkg/devspace/config/versions.Parse(0x1400055c2d0, {0x106a41100, 0x1082bbd20})
        /Users/runner/work/devspace/devspace/pkg/devspace/config/versions/versions.go:167 +0x420
github.com/loft-sh/devspace/pkg/devspace/config/loader.fillVariablesAndParse({0x1069f1b28, 0x140010b3860}, {0x106a16220, 0x1400067b280}, 0x1400055c2d0, {0x106a41100, 0x1082bbd20}, {0x10826be00, 0x17, 0x17})
        /Users/runner/work/devspace/devspace/pkg/devspace/config/loader/parser.go:134 +0xb0
github.com/loft-sh/devspace/pkg/devspace/config/loader.fillVariablesExcludeAndParse(...)
        /Users/runner/work/devspace/devspace/pkg/devspace/config/loader/parser.go:124
github.com/loft-sh/devspace/pkg/devspace/config/loader.(*commandsPipelinesParser).Parse(0x14001a87428, {0x1069f1b28, 0x140010b3860}, 0x140006177a0, 0x140005d62a0, {0x106a16220, 0x1400067b280}, {0x106a41100, 0x1082bbd20})
        /Users/runner/work/devspace/devspace/pkg/devspace/config/loader/parser.go:48 +0xf4
github.com/loft-sh/devspace/cmd.(*RawConfig).Parse(0x14000180190, {0x1069f1b28, 0x140010b3860}, 0x140006177a0, 0x140005d62a0, {0x106a16220, 0x1400067b280}, {0x106a41100, 0x1082bbd20})
        /Users/runner/work/devspace/devspace/cmd/root.go:359 +0xec
github.com/loft-sh/devspace/pkg/devspace/config/loader.(*configLoader).parseConfig(0x14000641430, {0x1069f1b28, 0x140010b3860}, 0x140006177a0, {0x106a314e0, 0x140003cfa40}, {0x0, 0x0}, {0x0, 0x0}, ...)
        /Users/runner/work/devspace/devspace/pkg/devspace/config/loader/loader.go:341 +0x35c
github.com/loft-sh/devspace/pkg/devspace/config/loader.(*configLoader).LoadWithParser(0x14000641430, {0x1069f1af0, 0x140001a3e00}, {0x0, 0x0}, {0x0, 0x0}, {0x1069b7780, 0x14000180190}, 0x14000180280, ...)
        /Users/runner/work/devspace/devspace/pkg/devspace/config/loader/loader.go:175 +0x6ac
github.com/loft-sh/devspace/cmd.parseConfig({0x106a35c38, 0x1082bc2f8})
        /Users/runner/work/devspace/devspace/cmd/root.go:326 +0x20c
github.com/loft-sh/devspace/cmd.BuildRoot({0x106a35c38, 0x1082bc2f8}, 0x0)
        /Users/runner/work/devspace/devspace/cmd/root.go:188 +0x17c
github.com/loft-sh/devspace/cmd.Execute()
        /Users/runner/work/devspace/devspace/cmd/root.go:132 +0x60
main.main()
        /Users/runner/work/devspace/devspace/main.go:20 +0xf0
```

**What did you expect to happen instead?**  

To give an error.

**How can we reproduce the bug?** (as minimally and precisely as possible)  

My devspace.yaml:
```
version: v2beta1
pipelines:
  after_dependency_background: |-
    on-after-deploy-container-ready-local
  pipelines:
  dev: |-
  ....
```

**Local Environment:**  
- DevSpace Version: 6.0.0
- Operating System: mac
- ARCH of the OS: ARM64
**Anything else we need to know?**  



<!-- DO NOT EDIT BELOW THIS LINE -->
/kind bug

--- a/pkg/devspace/config/versions/adjust.go
+++ b/pkg/devspace/config/versions/adjust.go
@@ -7,26 +7,74 @@ import (
 )
 
 func adjustConfig(config *latest.Config) error {
-	for name, v := range config.Vars {
-		v.Name = name
-	}
-	for name, command := range config.Commands {
-		command.Name = name
-	}
-	for name, pullSecret := range config.PullSecrets {
-		pullSecret.Name = name
-	}
-	for name, devPod := range config.Dev {
-		devPod.Name = name
-		for c, v := range devPod.Containers {
-			v.Container = c
+	if config.Vars != nil {
+		newObjs := map[string]*latest.Variable{}
+		for name, v := range config.Vars {
+			if v == nil {
+				continue
+			}
+			v.Name = name
+			newObjs[name] = v
 		}
+		config.Vars = newObjs
 	}
-	for name, pipeline := range config.Pipelines {
-		pipeline.Name = name
+	if config.Commands != nil {
+		newObjs := map[string]*latest.CommandConfig{}
+		for name, command := range config.Commands {
+			if command == nil {
+				continue
+			}
+			command.Name = name
+			newObjs[name] = command
+		}
+		config.Commands = newObjs
 	}
-	for name, dep := range config.Dependencies {
-		dep.Name = name
+	if config.PullSecrets != nil {
+		newObjs := map[string]*latest.PullSecretConfig{}
+		for name, pullSecret := range config.PullSecrets {
+			if pullSecret == nil {
+				continue
+			}
+			pullSecret.Name = name
+			newObjs[name] = pullSecret
+		}
+		config.PullSecrets = newObjs
+	}
+	if config.Dev != nil {
+		newObjs := map[string]*latest.DevPod{}
+		for name, devPod := range config.Dev {
+			if devPod == nil {
+				continue
+			}
+			devPod.Name = name
+			for c, v := range devPod.Containers {
+				v.Container = c
+			}
+			newObjs[name] = devPod
+		}
+		config.Dev = newObjs
+	}
+	if config.Pipelines != nil {
+		newObjs := map[string]*latest.Pipeline{}
+		for name, pipeline := range config.Pipelines {
+			if pipeline == nil {
+				continue
+			}
+			pipeline.Name = name
+			newObjs[name] = pipeline
+		}
+		config.Pipelines = newObjs
+	}
+	if config.Dependencies != nil {
+		newObjs := map[string]*latest.DependencyConfig{}
+		for name, dep := range config.Dependencies {
+			if dep == nil {
+				continue
+			}
+			dep.Name = name
+			newObjs[name] = dep
+		}
+		config.Dependencies = newObjs
 	}
 	if config.Images != nil {
 		newObjs := map[string]*latest.Image{}

